### PR TITLE
Cache engine redis cluster

### PR DESCRIPTION
**Pull request by nicosp**
Added support for using redis in cluster mode as a cache engine. Most code is identical but since Redis and RedisCluster are not covariant and we cannot simply extend RedisEngine so I moved the common code in a trait.

Still testing but I wanted to get some early feedback.

### CODE REVIEW

File: src/Cache/Engine/RedisClusterEngine.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+     * - `seeds` Seed nodes. Used when not using named cluster.
+     * - `timeout` timeout in seconds (float).
+     *
+     * @var array<string, mixed>
+     */
+    protected $_defaultConfig = [
+        'cluster' => null,
+        'seeds' => [],
+        'timeout' => 0,
+        'read_timeout' => 0,
```
Comment (markstory) [inline src/Cache/Engine/RedisClusterEngine.php]: Why mix snake case and camelback in options? Could we make this camelback like other options?

File: src/Cache/Engine/RedisEngineTrait.php [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         2.2.0
```
Comment (markstory) [inline src/Cache/Engine/RedisEngineTrait.php]:
```suggestion
 * @since         4.5.0
```

**Review by markstory — COMMENTED** [review]
Using a trait for the common methods is a reasonable approach to me. There are a few static analysis issues that will need to be fixed up, and tests will be required in order to merge this. Hopefully there is a way to run a small redis-cluster in github actions.

**Comment by ajibarra** [comment]
@nicosp hi, could you please take a look to failing checks? so it can be merged. thank you!

**Comment by nicosp** [comment]
Sorry for the lack of progress here. We are currently busy upgrading to CakePHP 5.0. We will resume development after we upgrade to 5.

**Comment by ADmad** [comment]
New features need to target `5.next` now and since changing the target branch will cause merge conflicts I am closing this for now. Feel free to reopen a new PR targeting `5.next` in the future.

**Pull request closed without merging**